### ISSUE TITLE
Is there any way of showing empty space at last page when there aren't enough items to fill it?

### ISSUE THREAD

**Issue by djmarcos88** [OwlCarousel2/OwlCarousel2#1912]
So, I'm currently having a discussion with my client because they want to display the empty space on the last page if there are not enough items to fill it. I don't think it looks good, but anyway, it's what they want.

The problem is that I don't think there's a way to make OwlCarousel do this.
Say you slide by 2 items, and have 5 items...the last page should show the 5th item and an "empty" 6th item. Just so that if the items are similar users won't get confused on what they are looking at.

**Comment by keefyhub** [comment]
you could check the amount of items in the carousel and then append some extra items before the carousel loads?

something like this - https://codepen.io/keefyboooo/pen/eROdWK

Its a bit hacky, but clients!

**Comment by djmarcos88** [comment]
Yeah, that was my first idea. But due to how our carousel looks on mobile it's not that straightforward, I was looking for a more automatic solution.
I guess I could do something similar but having to validate what to add on every resize

**Comment by keefyhub** [comment]
I guess an automatic compromise might be to use `center` - https://owlcarousel2.github.io/OwlCarousel2/demos/center.html

It wont give you exactly what you need, but might be enough to please the client.

**Comment by fenicento** [comment]
+1
Same situation, client wants the last page to be full, with white space if there are not enough items. Could you point me to the part of code that has to be modified in order to achieve this?

**Comment by djmarcos88** [comment]
Well, I've actually convinced my client to not do this by making the sliding animation slower.

But if you need to do this you can code something similar to what keefyhub shows on his first example. Just append extra empty items depending on the number of real items you have. That way you "complete" the last page.

Bear in mind that if you show a different numbers of items in small resolutions this will be a problem.

**Comment by fenicento** [comment]
I thought about It, but It seems a bit rough to me. Also, it's annoying to
check for different resolutions and resize and add/remove items
accordingly. It's much more consistent to me to just change the logic of
page browsing. I also think It would be handy to have an option for that,
since I'm not the only one requiring the feature.

Il ven 16 giu 2017, 18:53 Marcos Djivelekian <notifications@github.com> ha
scritto:

**Comment by fenicento** [comment]
if anyone is interested, I managed to achieve full page navigation by modifying the following code (line 1048 of owl.carousel.js):
`
else {
			maximum = this._items.length - settings.items;
		}
`

becomes:
`
else {
			maximum = this._items.length;
		}
`
I'm testing for side effects but it seems to be working fine.

**Comment by djmarcos88** [comment]
Nice! It would be cool to have an option to change this as needed.
